src/codegen.cpp

```diff
@@ -533,6 +533,51 @@ static AttributeList get_attrs_zext(LLVMContext &C)
                 {Attributes(C, {Attribute::ZExt})});
 }
 
+static AttributeList get_attrs_ipoeffects(LLVMContext &C, jl_code_instance_t *ci)
+{
+    uint32_t e = ci->ipo_purity_bits;
+    uint8_t consistent = (e >> 0) & 0x07;
+    uint8_t effect_free = (e >> 3) & 0x03;
+    uint8_t nothrow = (e >> 5) & 0x01;
+    uint8_t terminates = (e >> 6) & 0x01;
+    uint8_t notaskstate = (e >> 7) & 0x01;
+    uint8_t inaccessiblememonly = (e >> 8) & 0x03;
+    uint8_t nonoverlayed = (e >> 10) & 0x01;
+#if JL_LLVM_VERSION >= 140000
+    AttrBuilder attr(C);
+#else
+    AttrBuilder attr;
```

Comment:
Perhaps we could prefix all of the custom string attributes with `julia.` to avoid any namespace collisions

Reply:
Yeah, they are there mostly for debugging, we probably want a smarter way to encode this